### PR TITLE
Add Connery Tool and Toolkit

### PR DESCRIPTION
**Pull request by machulav**
## Summary

This PR implements the "Connery Actions Tool" and "Connery Actions Toolkit". Using them, you can integrate Connery actions into your LangChain agents and chains.

Connery is an open-source plugin infrastructure for AI.

With Connery, you can easily create a custom plugin, which is essentially a set of actions, and use them in your LangChain agents and chains. Connery will handle the rest: runtime, authorization, secret management, access management, audit logs, and other vital features. Also, you can find a lot of ready-to-use plugins from our community.

Learn more about Connery:

- GitHub repository: https://github.com/connery-io/connery-platform
- Documentation: https://docs.connery.io
- Twitter: https://twitter.com/connery_io

## TODOs

- [x] Connery Actions Tool
   - [x] Docs
   - [x] Example
- [x] Connery Actions Toolkit
  - [x] Docs
  - [x] Example
- [x] Linting
- [x] Formatting

### CODE REVIEW

File: langchain/src/tools/index.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export { SerpAPI, type SerpAPIParameters } from "./serpapi.js";
 export { DadJokeAPI } from "./dadjokeapi.js";
 export { BingSerpAPI } from "./bingserpapi.js";
 export { Tool, type ToolParams, StructuredTool } from "./base.js";
+export { ConneryService } from "./connery.js";
```
Comment (jacoblee93) [inline langchain/src/tools/index.ts:5]: Remove and then you'll need to create new entrypoints for the new tool + toolkit to get exports working properly:

https://github.com/langchain-ai/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md

File: langchain/src/agents/toolkits/connery/connery.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   * @param conneryService The ConneryService instance.
+   * @returns A Promise that resolves to a ConneryToolkit instance.
+   */
+  static async createInstance(
+    conneryService: ConneryService
+  ): Promise<ConneryToolkit> {
+    const toolkit = new ConneryToolkit();
+    toolkit.tools = [];
+
+    const actions = await conneryService.listActions();
```
Comment (jacoblee93) [inline langchain/src/agents/toolkits/connery/connery.ts:30]: Neat!

File: langchain/src/tools/connery.ts [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    const apiKey =
+      params?.apiKey ?? getEnvironmentVariable("CONNERY_RUNNER_API_KEY");
+
+    if (!runnerUrl || !apiKey) {
+      throw new Error(
+        "CONNERY_RUNNER_URL and CONNERY_RUNNER_API_KEY environment variables must be set."
+      );
+    }
+
+    this.runnerUrl = runnerUrl;
```
Comment (jacoblee93) [inline langchain/src/tools/connery.ts:213]: Is there a default value that makes sense for this?

**Review by jacoblee93 — COMMENTED** [review]
Nice! Only comment is to please use unique entrypoints for the new modules.

File: langchain/src/tools/connery.ts (hunk 3 above)
Comment (machulav) [reply langchain/src/tools/connery.ts:213]: There is no default value. The idea of Connery is that everyone has their own isolated runner with a unique URL where they keep all their actions and connections.

**Comment by machulav** [comment]
Still have trouble with the docs, though:

```
Compiled with problems:

ERROR in ../../examples/src/agents/connery_mrkl.ts
Module build failed (from ./code-block-loader.js):
Error: Could not find docs for agents_toolkits_connery.ConneryToolkit or schema_agents_toolkits_connery.ConneryToolkit in api_refs/public/
    at /langchainjs/docs/core_docs/code-block-loader.js:97:15
    at Array.forEach (<anonymous>)
    at Object.webpackLoader (/langchainjs/docs/core_docs/code-block-loader.js:90:13)
```

The `docs/api_refs` build seems fine. I was able to build it separately.
So it seems I've made a mistake in the entry points.

Do you know what could it be?

**Comment by machulav** [comment]
Following the SQL toolkit as an example, I renamed the file from `langchain/src/agents/toolkits/connery/connery.ts` to `langchain/src/agents/toolkits/connery/index.ts` in the last commit, and it fixed the issue locally. 

However, there are still some issues with the docs deployment, which I can not see.

**Comment by jacoblee93** [comment]
Thanks! Yeah we've been mucking with the build a little bit to speed it up and had some issues where it passed due to caching, you might have pulled in at the wrong time. I'll fix it.

File: examples/src/agents/connery_mrkl.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+const model = new OpenAI({ temperature: 0 });
+const conneryService = new ConneryService();
+const conneryToolkit = await ConneryToolkit.createInstance(conneryService);
+
+const executor = await initializeAgentExecutorWithOptions(
+  conneryToolkit.tools,
+  model,
+  {
+    agentType: "zero-shot-react-description",
```
Comment (jacoblee93) [inline examples/src/agents/connery_mrkl.ts:30]: Not blocking for this but the OpenAI Functions agent probably has the best performance right now

**Comment by jacoblee93** [comment]
I am not able to push to your branch but can you merge `main`?

File: examples/src/agents/connery_mrkl.ts (hunk 4 above)
Comment (machulav) [reply examples/src/agents/connery_mrkl.ts:30]: Good point! I will add another example later.

**Comment by machulav** [comment]
Done

**Pull request merged**